### PR TITLE
 ENH: resources: Teach get/put to infer destination basename 

### PR DESCRIPTION
**Pull request by kyleam**
This mostly addresses #295, but it looks like the particular example that issue gave still won't work because `AwsEc2` doesn't define a `get()` method.

---

- [x] fix failing tests

### CODE REVIEW

File: niceman/resource/session.py [hunk 1]
```diff
@@ -267,31 +269,71 @@ def niceman_exec(self, command, args):
...
+            exists = op.exists
+            mkdir = os.makedirs
+            # Avoid a direct `abspath` call to preserve trailing slash.
+            dest_path = op.join(op.abspath(op.dirname(dest_path or "")),
+                                op.basename(dest_path or ""))
+        elif dest_path and op.isabs(dest_path):
+            exists = self.exists
+            mkdir = partial(self.mkdir, parents=True)
+        else:
+            raise ValueError("Remote destination paths must be absolute")
```
Comment (kyleam) [inline niceman/resource/session.py]: I've assumed incorrectly here:

https://travis-ci.org/ReproNim/niceman/jobs/434716977#L1349

File: niceman/resource/session.py [hunk 2]
```diff
@@ -267,31 +269,74 @@ def niceman_exec(self, command, args):
...
+        -------
+        The path to the destination, possibly adjusted to add the basename.
+        """
+        if local:
+            exists = op.exists
+            mkdir = os.makedirs
+        else:
+            exists = self.exists
+            mkdir = partial(self.mkdir, parents=True)
+
```
Comment (mjtravers) [inline niceman/resource/session.py]: Interesting way to unclutter your if statements below by setting up your function calls here. Will remember the partial() trick.

**Comment by yarikoptic** [comment]
looks good to me, added just a minor sugaring in 7ccdfee94c0d381a962ad4bc97f9768b864f069e

**Comment by kyleam** [comment]
> looks good to me, added just a minor sugaring in 7ccdfee

and you addition looks good to me, thanks

**Pull request merged**